packages/platform-browser/src/IndexedDbQuery.ts

```diff
@@ -0,0 +1,418 @@
+    Table extends IndexedDbTable.IndexedDbTable.TableName<
+      IndexedDbVersion.IndexedDbVersion.Tables<Source>
+    > = never
+  > {
+    readonly [TypeId]: TypeId
+    readonly tables: HashMap.HashMap<string, IndexedDbVersion.IndexedDbVersion.Tables<Source>>
+    readonly table: Table
+    readonly database: globalThis.IDBDatabase
+    readonly IDBKeyRange: typeof globalThis.IDBKeyRange
+    readonly transaction?: globalThis.IDBTransaction
+
+    readonly select: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Select<Source, Table, Index>
+      (): Select<Source, Table, never>
+    }
+
+    readonly count: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Count<Source, Table, Index>
+      (): Count<Source, Table, never>
+    }
+
+    readonly delete: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Delete<Source, Table, Index>
+      (): Delete<Source, Table, never>
+    }
+
+    readonly insert: (value: internal.ModifyWithKey<Source, Table>) => Modify<Source, Table>
+    readonly insertAll: (values: Array<internal.ModifyWithKey<Source, Table>>) => ModifyAll<Source, Table>
+    readonly upsert: (value: internal.ModifyWithKey<Source, Table>) => Modify<Source, Table>
+    readonly upsertAll: (values: Array<internal.ModifyWithKey<Source, Table>>) => ModifyAll<Source, Table>
+    readonly clear: Clear<Source, Table>
+  }
+
+  /**
+   * @since 1.0.0
+   * @category models
+   */
+  export interface ClearAll<
+    Source extends IndexedDbVersion.IndexedDbVersion.AnyWithProps = never
+  > {
+    [Symbol.iterator](): Effect.EffectGenerator<Effect.Effect<void>>
+
+    readonly [TypeId]: TypeId
+    readonly tables: HashMap.HashMap<string, IndexedDbVersion.IndexedDbVersion.Tables<Source>>
+    readonly database: globalThis.IDBDatabase
+    readonly transaction?: globalThis.IDBTransaction
+  }
+
+  /**
+   * @since 1.0.0
+   * @category models
+   */
+  export interface Clear<
+    Source extends IndexedDbVersion.IndexedDbVersion.AnyWithProps = never,
+    Table extends IndexedDbTable.IndexedDbTable.TableName<
+      IndexedDbVersion.IndexedDbVersion.Tables<Source>
+    > = never
+  > {
+    [Symbol.iterator](): Effect.EffectGenerator<Effect.Effect<void>>
+
+    readonly [TypeId]: TypeId
+    readonly from: From<Source, Table>
+  }
+
+  /**
+   * @since 1.0.0
+   * @category models
+   */
+  export interface Count<
+    Source extends IndexedDbVersion.IndexedDbVersion.AnyWithProps = never,
+    Table extends IndexedDbTable.IndexedDbTable.TableName<
+      IndexedDbVersion.IndexedDbVersion.Tables<Source>
+    > = never,
+    Index extends IndexFromTable<Source, Table> = never
+  > {
+    [Symbol.iterator](): Effect.EffectGenerator<Effect.Effect<number>>
+
+    readonly [TypeId]: TypeId
+    readonly from: From<Source, Table>
+    readonly index?: Index
+    readonly only?: internal.ExtractIndexType<Source, Table, Index>
```

Comment:
You can't use internal types in public modules

Reply:
I moved the internal types inside the `namespace`